x/evm/keeper/state_transition.go

```diff
@@ -145,19 +145,21 @@ func (k *Keeper) ApplyTransaction(tx *ethtypes.Transaction) (*types.MsgEthereumT
 	res.Hash = txHash.Hex()
 
 	// Set the bloom filter and commit only if transaction is NOT reverted
+	logs := k.GetTxLogs(txHash)
+	// Commit and switch to original context
+	if !res.Reverted {
+		commit()
+	}
+	k.ctx = originalCtx
+
+	// refund gas prior to handling the vm error in order to set the updated gas meter
 	if !res.Reverted {
```

Comment:
I still don't understand why this only applies to non-reverted txs. Can you add a comment?